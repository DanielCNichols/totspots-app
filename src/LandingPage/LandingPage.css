.landingpage_hero {
  background-color: lightblue;
  text-align: center;
  height: 150px;
  margin: 45px auto;
  padding-top: 30px;
}

.landingpage {
  margin-bottom: 100px;
}

header > h1 {
  margin: 0;
}

.landingpage p {
  text-align: center;
}
.search p {
  text-align: center;
}

.search_form_label {
  text-align: center;
  display: block;
  margin: 1.5em;
}
.search_form {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
}

.search_form_text {
  display: block;
  width: 80%;
  margin: 5px auto;
  padding: 5px 10px;
  height: 2.5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}

.search_form_select { 
  display: block;
  width: 80%;
  padding: 5px 10px;
  height: 2.5rem;
  border-radius: 5px;
  margin: 5px auto;
  cursor: pointer;
}

.search_form_button {
  display: block; 
  margin: 10px auto;
  border-radius: 17px;
  background-color: #4392F1;
  width: 50%;
  height: 40px;
  font-size: 1.1em;
}

.landingpage a:visited{
  color: black;
}

@media screen and (min-width: 450px) {
  .landingpage {
    margin: auto;
  }

  .landingpage_hero {
    margin-top: 45px;
    height: 200px;
  }
  
  .search_form {
    width: 70%;
  }

  .search_form_button {
    width: 30%;
  }

  .search_form_select, .search_form_text {
    width: 70%;
  }
}




@media screen and (min-width: 800px) {
  .search {
    width: 50%;
    margin: 0 auto 50px auto;
  }

  .search_form_select, .search_form_text {
    min-width: 220px;
  }

  .search_form_button {
    min-width: 180px;
  }
  
  .landingpage {
    width: 100%;
    font-size: 1.3em;
  }



  .landingpage_hero {
    margin-top: 20px;
    height: 250px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    width: 100%;
  }

  .landingpage_hero h1 {
    font-size: 2em;
  }
}