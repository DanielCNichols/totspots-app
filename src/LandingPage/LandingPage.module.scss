* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

// Mobile styling

.wrapper {
  background-color: #c4dcf8;
}

.form {
  display: flex;
  flex-direction: column;
  width: 90%;
  position: absolute;
  top: 75%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.form > input,
select,
button {
  margin: 10px 0;
  height: 45px;
}

input,
select {
  padding: 10px;
  font-size: 1.2em;
}

.form > button {
  background-color: #4a6ddd;
  color: white;
  font-size: 1.3em;
}

.heroImage {
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
    url('../assets/annie-spratt-WS2anBsrum0-unsplash.jpg');
  height: 600px;
  width: 100%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  z-index: 0;
  display: flex;
}

.heroText {
  text-align: center;
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
}

.onboarding {
  display: flex;
  flex-direction: column;
  margin: 40px auto 0 auto;
  padding-bottom: 100px;
}

.onboardingHeader {
  text-align: center;
  margin: 25px auto;
  width: 90%;
  word-wrap: break-word;
}

.cardContainer {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 10px auto;
  text-align: center;
  justify-content: space-evenly;
}

.onboardingCard {
  width: 100%;
  margin: 25px 0;
  // box-shadow: 0px 0px 2px 2px rgba(0, 0, 0, 0.75);
  // padding: 25px;
}

.onboardingCardSvg {
  width: 100%;
  height: 150px;
  margin: 5% auto;
  color: #4881c2be;
}

h2,
h4 {
  text-align: center;
  margin: 10px;
}

.onboardingContent {
  grid-row-start: 2;
}

@media screen and (min-width: 450px) {
  .wrapper {
    background-color: #c4dcf8;
  }

  .heroImage {
    background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
      url('../assets/annie-spratt-WS2anBsrum0-unsplash.jpg');
    height: 500px;
    width: 100%;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    position: relative;
    z-index: 0;
    display: flex;
  }

  .heroText {
    text-align: center;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: white;
  }

  .form {
    z-index: 1;
    width: 60%;
    position: absolute;
    top: 80%;
    left: 50%;
    transform: translate(-50%, -50%);
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  .form > input,
  select {
    width: 300px;
    height: 45px;
    font-size: 18px;
    padding: 0 10px;
    border-radius: 5px;
    border: none;
    margin: 0 10px;
  }

  .form > button {
    width: 75px;
    height: 45px;
    font-size: 18px;
    border: none;
    border-radius: 5px;
    color: white;
    background-color: #4a6ddd;
  }

  .onboarding {
    width: 70%;
    margin: 100px auto 0 auto;
    padding-bottom: 45px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
  }
}

@media screen and (min-width: 800px) {
  .form {
    width: 80%;
  }

  .onboarding {
    margin: 50px auto 0 auto;
  }

  .onboardingHeader {
    margin-bottom: 75px;
  }

  .cardContainer {
    flex-direction: row;
    width: 100%;
    justify-content: space-between;
  }

  .onboardingCard {
    width: 30%;
    padding: 20px;
  }
}
