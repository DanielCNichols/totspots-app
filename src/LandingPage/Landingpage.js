import React from 'react';
import styles from './LandingPage.module.scss';
import { FaGlassMartini, FaCoffee } from 'react-icons/fa';
import { GiKnifeFork } from 'react-icons/gi';
import AutoCompleteInput from '../AutocompleteInput/AutocompleteInput';
export default function newLanding() {
  function handleSelect(place) {
    console.log(place);
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.heroImage}>
        <div className={styles.heroText}>
          <h1>For parents, by parents!</h1>
          <p>Find!</p>
        </div>
        <div className={styles.form}>
          <select>
            <option value="">Find a...</option>
            <option value="">restaurant</option>
            <option value="">bar</option>
            <option value="">coffee shop</option>
            <option value="">museum</option>
          </select>
          {/* <input type="text" placeholder="in..." /> */}
          <AutoCompleteInput onPlaceLoaded={handleSelect} />
          <button type="submit">Go!</button>
        </div>
      </div>
      <div className={styles.onboarding}>
        <div className={styles.onboardingHeader}>
          <h3 style={{ fontSize: '2.0em' }}>
            SWEET! a review app for parents
          </h3>
          <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit.
            Necessitatibus maxime eveniet laudant.
          </p>
        </div>
        <div className={styles.cardContainer}>
          <div className={styles.onboardingCard}>
            <FaGlassMartini
              className={styles.onboardingCardSvg}
            ></FaGlassMartini>
            <div className={styles.onboardingContent}>
              <h4>Find!</h4>
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque
                laudantium delectus obcaecati deleniti quasi dolorum,
                perferendis tempore
              </p>
            </div>
          </div>
          <div className={styles.onboardingCard}>
            <FaCoffee className={styles.onboardingCardSvg}></FaCoffee>
            <div className={styles.onboardingContent}>
              <h4>Review!</h4>
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque
                laudantium delectus obcaecati deleniti quasi dolorum,
                perferendis tempore
              </p>
            </div>
          </div>
          <div className={styles.onboardingCard}>
            <GiKnifeFork className={styles.onboardingCardSvg}></GiKnifeFork>
            <div className={styles.onboardingContent}>
              <h4>Enjoy!</h4>
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque
                laudantium delectus obcaecati deleniti quasi dolorum,
                perferendis tempore
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
