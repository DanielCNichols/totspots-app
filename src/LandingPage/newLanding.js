import React from 'react';
import styles from './newLanding.module.scss';

export default function newLanding() {
  return (
    <div className={styles.heroImage}>
      <div className={styles.heroText}>
        <h1>This is a header</h1>
        <p>This is some text</p>
      </div>
      {/* <div className={styles.form}>
        <select>
          <option value="">Find a...</option>
          <option value="">restaurant</option>
          <option value="">bar</option>
          <option value="">coffee shop</option>
          <option value="">museum</option>
        </select>
        <input type="text" placeholder="in..." />
        <button type="submit">Go!</button>
      </div> */}
    </div>
  );
}
