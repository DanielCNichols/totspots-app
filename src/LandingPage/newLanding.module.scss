.heroImage {
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
    url('../assets/annie-spratt-WS2anBsrum0-unsplash.jpg');
  height: 600px;
  width: 100%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  z-index: 0;
  display: flex;
}

.heroText {
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
}

.form {
  z-index: 1;
  position: absolute;
  top: 80%;
  left: 27.5%;
  display: flex;
}

.form > input,
select {
  width: 300px;
  height: 45px;
  font-size: 18px;
  padding: 0 10px;
  border-radius: 5px;
  border: none;
  margin: 0 10px;
}

.form > button {
  width: 75px;
  height: 45px;
  font-size: 18px;
  border: none;
  border-radius: 5px;
  color: white;
  background-color: #4a6ddd;
}
