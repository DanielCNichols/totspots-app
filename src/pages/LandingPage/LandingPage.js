import React from 'react';
import Hero from '../../components/Hero';
import landingImage from '../../assets/index';
import SearchForm from '../../components/SearchForm';
import Footer from '../../components/Footer';
import {FaGlassMartini, FaCoffee} from 'react-icons/fa';
import {GiKnifeFork} from 'react-icons/gi';
import s from '../../sass/layout/LandingPage.module.scss';

export default function LandingPage(props) {
  const handleSearch = (lat, lng, type) => {
    props.history.push(`/results/search?type=${type}&lat${lat}&lng=${lng}`);
  };

  return (
    <section className={s.LandingPage}>
      <Hero imgSource={landingImage}>
        <header>
          <h1>Totpots</h1>
          <p>For parents, by parents</p>
        </header>
        <SearchForm onSearch={handleSearch} />
      </Hero>
      <section className={s.onBoarding}>
        <header>
          <h2>Finally, a review app for parents!</h2>
        </header>
        <div className={s.cardContainer}>
          <div className={s.card}>
            <FaGlassMartini />
            <div className={s.cardContent}>
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam,
                accusamus voluptatum obcaecati cupiditate illum delectus
                blanditiis molestias voluptate reprehenderit non culpa ea
                mollitia, reiciendis debitis quo dolores sit quam! Adipisci!
              </p>
            </div>
          </div>
          <div className={s.card}>
            <FaCoffee />
            <div className={s.cardContent}>
              <p>
                Lorem, ipsum dolor sit amet consectetur adipisicing elit.
                Incidunt aspernatur molestias eius fuga reprehenderit, placeat
                ullam eos magni unde optio quos minima corrupti. Ullam veniam
                animi dolor, ducimus minus numquam!
              </p>
            </div>
          </div>
          <div className={s.card}>
            <GiKnifeFork />
            <div className={s.cardContent}>
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Rerum
                distinctio maiores dolore sit ratione eveniet, quisquam
                voluptates perferendis natus excepturi non unde, accusantium
                delectus aperiam rem repellat laudantium totam recusandae.
              </p>
            </div>
          </div>
        </div>
      </section>
      <Footer />
    </section>
  );
}
