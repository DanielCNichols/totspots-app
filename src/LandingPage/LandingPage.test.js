import React from 'react';
import ReactDOM from 'react-dom'
import LandingPage from './LandingPage.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders LandingPage without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <LandingPage />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders LandingPage as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <LandingPage name="LandingPage"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
