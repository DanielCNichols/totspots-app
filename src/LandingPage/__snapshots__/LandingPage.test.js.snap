// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LandingPage as expected 1`] = `
<section
  className="landingpage"
>
  <header
    className="landingpage_hero"
  >
    <h1>
      Tot Spots
    </h1>
    <p>
      For parents, by parents
    </p>
  </header>
  <div
    className="search"
  >
    <p>
      I want to take my family to a...
    </p>
    <form
      className="search_form"
      onSubmit={[Function]}
    >
      <label
        className="search_form_label"
        htmlFor="type"
        required={true}
      >
        Venue or event...
        <select
          aria-label="choose Venue or event"
          aria-required="true"
          className="search_form_select"
          id="type"
          name="type"
        >
          <option
            value=""
          >
            Select a type of venue/event
          </option>
          <option
            value="Restaurant"
          >
            Restaurants
          </option>
          <option
            value="Bar"
          >
            Brewery/Bar
          </option>
          <option
            value="Coffee"
          >
            Coffee Shop
          </option>
          <option
            value="Educational"
          >
            Educational
          </option>
          <option
            value="music"
          >
            Music
          </option>
          <option
            value="outdoor"
          >
            Outdoor
          </option>
        </select>
      </label>
      <p>
        in...
      </p>
      <label
        className="search_form_label"
        htmlFor="city"
      >
         
        City
        <input
          aria-label="choose city"
          aria-required="true"
          className="search_form_text"
          id="city"
          name="city"
          placeholder="Durham"
          required={true}
          type="text"
        />
      </label>
      <label
        className="search_form_label"
        htmlFor="state"
      >
         
        State
        <input
          aria-label="Choose state"
          aria-required="true"
          className="search_form_text"
          max-length="2"
          name="state"
          placeholder="NC"
          required={true}
          type="text"
        />
      </label>
      <div
        className="search_form_control"
      >
        <button
          className="search_form_button"
          type="submit"
        >
          Let's go!
        </button>
      </div>
    </form>
  </div>
  <p>
    Have an account? Go ahead and 
    <a
      href="/login"
      onClick={[Function]}
    >
      log in!
    </a>
  </p>
</section>
`;
